### ISSUE TITLE
pypy 7.3.16 (Python 3.10.14) is missing mmap.MAP_POPULATE

### ISSUE THREAD

**Issue by nE0sIghT** [pypy/pypy#4991]
According to docs [1] `MAP_POPULATE` was added in the Python 3.10, however it is missing in the PyPy
```
>>>> from mmap import MAP_POPULATE
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: cannot import name 'MAP_POPULATE' from 'mmap' (unknown location)
```

Reproduced with the `docker.io/pypy:3.10` image. Here is `python --version`:

```
Python 3.10.14 (75b3de9d9035, Apr 21 2024, 10:54:48)
[PyPy 7.3.16 with GCC 10.2.1 20210130 (Red Hat 10.2.1-11)]
```

P.S.: this is needed for the apt-mirror2 [2]

[1] https://docs.python.org/3/library/mmap.html#mmap.MAP_POPULATE
[2] https://gitlab.com/apt-mirror2/apt-mirror2/-/merge_requests/126

**Comment by mattip** [comment]
Also missing: `MADV_FREE`

**Comment by mattip** [comment]
about `MADV_FREE`: there is some logic around whether it is defined or not, which makes exporting it in moduledef.py more complicated.

**Comment by mattip** [comment]
What does apt-mirror do on windows and macos where `MAP_POPULATE` is not available?

**Comment by mattip** [comment]
Fixed in e617a648325

**Comment by nE0sIghT** [comment]
> What does apt-mirror do on windows and macos where MAP_POPULATE is not available?

It just doesn't support Windows/MacOS yet :-)
